sdk/storage/Azure.Storage.Blobs/src/BlobExtensions.cs

```diff
@@ -142,5 +144,39 @@ internal static string ToTagsString(this Tags tags)
                 ExpiresOn = properties.ExpiresOn,
                 IsSealed = properties.IsSealed,
             };
+
+        /// <summary>
+        /// Internal. Parses Object Replication Policy ID from Rule ID and sets the Policy ID.
+        /// </summary>
+        /// <returns></returns>
+        internal static IDictionary<string, IDictionary<string, string>> ParseObjectReplicationIds(IDictionary<string, string> OrIds)
```

Comment:
This technically isn't extension method. Is there a better place to host it ?

Reply:
I'm also not sure where else to put this method. It will get used by BlobProperties, BlobDownloadDetails, and eventually the list blob item properties. @seanmcc-msft any suggestions?

Reply:
@amnguye, if you change the signature to
internal static IDictionary<string, IDictionary<string, string>> ParseObjectReplicationIds(
**this** IDictionary<string, string> OrIds)

It will be an extension method.